test/test.py

```diff
@@ -366,3 +366,21 @@ def map_object(self, obj):
         # as a context-manager so the caller does not retain the path for
         # later access.
         yield path
+
+    def copy_source_data(self, target, source):
+        """Copy the cached sources for a `source` to `target`
+
+        This will copy all the downloaded data for a specified source
+        to the `target` directory, with a folder structure in a way so
+        it can be used to initialize the cache, via the constructor's
+        `cache_from` argument.
+        """
+        from_path = os.path.join(self._cachedir, "sources", source)
+        assert os.access(from_path, os.R_OK)
```

Comment:
Wouldn't you want to bail out if the directory does not exist? You call it unconditionally, so any test that uses other sources than the one you copy will not have the sources-directory, right? Maybe as simple as `if not os.path.isdir(from_path): return`?

Reply:
Very good point!